snapcraft/internal/lifecycle/_runner.py

```diff
@@ -209,7 +209,9 @@ def _run_step(self, step, part, part_names):
 
         part = _replace_in_part(part)
 
-        getattr(part, step)()
+        is_classic = self.config.data['confinement'] == 'classic'  # type: bool
+
+        getattr(part, step)(is_classic=is_classic)
```

Comment:
For some reason adding kwargs to all steps just to handle `is_classic` in `prime()` is bothering me. It's not something about the snap that will change, it's a fundamental property of it. It seems like something that would just be generally known, like its name. Would it be weird to put this in the `ProjectOptions` instead, or otherwise make it generally available? Maybe even the `PluginHandler`'s `__init__`.

Reply:
ok, I'll go down that path (`project_options`)

Reply:
I ended up adding it to the PluginHandler